src/SubjectPreview.js

```diff
@@ -1,61 +1,63 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import {makeStyles} from "@material-ui/core/styles";
+import React from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
 import ListItem from "@material-ui/core/ListItem";
 import ListItemText from "@material-ui/core/ListItemText";
 import Link from "@material-ui/core/Link";
 import Box from "@material-ui/core/Box";
 import Grid from "@material-ui/core/Grid";
 import Typography from "@material-ui/core/Typography";
+import Checkbox from "@material-ui/core/Checkbox";
 
 import QuantityIndicator from "./QuantityIndicator";
-import {daysInWeek, maxcredits, maxlength} from "./constants";
-
-
-const useStyles = makeStyles((theme) => ({
-    root: {
-        display: "block"  // This overrides horizontal layout of ListItem's children
-    },
-}));
+import { daysInWeek, maxcredits, maxlength } from "./constants";
+import { toggleSubject } from "./redux/actions";
 
 
 /**
  * Displays brief information about the subject organized in a small grid.
  */
-const SubjectPreview = ({ subject, onShowMore }) => {
-    const classes = useStyles()
-
+export const SubjectPreview = ({ subject, onShowMore, selected = false, onToggleSelect }) => {
```

Comment:
Je tady ten default argument nutný?

Reply:
K ničemu neslouží no (maličko to je k něčemu v testech), takže asi ne. Já jsem tam čaroval s tím, jestli značit ty callbacky onShowMore a onToggleSelect jako povinné a spolu s tím jsem vymyslel, že selected by po vzoru Checkboxu mohlo být taky takhle s defaultním false. Jsou to takový blbosti/detaily, příště už to nebudu komplikovat 😄